common/lib/common-utils/src/fluidError.ts

```diff
@@ -0,0 +1,66 @@
+/*!
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import {
+    IFluidError,
+    ITelemetryProperties,
+    ISensitiveDebugData,
+    isIFluidError,
+} from "@fluidframework/common-definitions";
+
+export abstract class FluidError extends Error implements IFluidError {
+    public abstract errorType: string;
+
+    private props: ITelemetryProperties = {};
+    private debugData: ISensitiveDebugData = {};
+    constructor(
+        message: string,
+        props: ITelemetryProperties = {},
+        debugData: ISensitiveDebugData = {},
+    ) {
+        super(message);
+        this.addDetails(props, debugData);
+    }
+
+    public getFluidTelemetryProps() { return { ...this.props, message: this.message, errorType: this.errorType }; }
+    public getSensitiveDebugData() {
+        return { ...this.debugData, stack: this.stack ?? "" };
+    } //* Or implement deep copy? Not even possible...?
+
+    public addDetails(props: Partial<ITelemetryProperties>, debugData: Partial<ISensitiveDebugData>) {
+        this.props = { ...this.props, ...props };
+        this.debugData = { ...this.debugData, ...debugData };
+    }
+}
+
+export class GenericFluidError extends FluidError {
+    public errorType: string = "generic";
+}
+
+export class ExternalFluidError extends FluidError {
+    public errorType: string = "external";
+
+    constructor(err: any) {
+        //* start with promoting err's message, per present behavior, and then pull out in later scoped change
+        super("External Error");
+
+        this.addDetails({}, { innerError: err });
+        if (err.stack !== undefined && err.stack !== "") {
```

Comment:
Note: err could be a string, or a number, or even undefined. This code will explode, prior code handled any types